app/controllers/katello/api/v2/content_export_incrementals_controller.rb

```diff
@@ -0,0 +1,99 @@
+                                                 "If not provided the most recent export history will be used."), :required => false
+    def library
+      tasks = async_task(::Actions::Pulp3::Orchestration::ContentViewVersion::ExportLibrary,
+                          @organization,
+                          destination_server: params[:destination_server],
+                          chunk_size: params[:chunk_size_mb],
+                          from_history: @history)
+      respond_for_async :resource => tasks
+    end
+
+    private
+
+    def find_exportable_content_view_version
+      @version = ContentViewVersion.exportable.find_by_id(params[:id])
+      throw_resource_not_found(name: 'content view version', id: params[:id]) if @version.blank?
+      @view = @version.content_view
+    end
+
+    def find_library_export_view
+      @view = ::Katello::ContentView.find_library_export_view(destination_server: params[:destination_server],
+                                                                organization: @organization,
+                                                                create_by_default: false)
+      if @view.blank?
+        msg = _("Unable to incrementally export. Do a Full Export the library content "\
+                "before updating from the latest increment.")
+        fail HttpErrors::BadRequest, msg
+      end
+    end
+
+    def find_history
+      if !params[:from_history_id].blank?
+        @history = ::Katello::ContentViewVersionExportHistory.find(params[:from_history_id])
+        if @history.blank?
+          throw_resource_not_found(name: 'content view version export history',
+                                   id: params[:from_history_id])
+        end
+      else
+        @history = ::Katello::ContentViewVersionExportHistory.
+                      pick_recent_history(@view,
+                                          destination_server: params[:destination_server])
+        if @history.blank?
+          if params[:action] == :library
+            msg = _("Prior content view export history belonging to the library export content view was not found "\
+                    "in organization with id %{id} and destination_server %{server}. "\
+                    "Please choose the full export.") %
+                    {id: @organization.id, server: params[:destination_server]}
+          else
+            msg = _("Prior content view export history belonging to the content view '%{view}' was not found "\
+                    " for the destination_server '%{server}'. Please choose the full export.") %
+                    {view: @view.name, server: params[:destination_server]}
+          end
+          fail HttpErrors::NotFound, msg
+        end
+      end
+    end
+
+    def find_exportable_organization
+      find_organization
+      unless @organization.can_export_library_content?
+        throw_resource_not_found(name: 'organization', id: params[:organization_id])
```

Comment:
Is this the right error to throw? We already found the org at this point

Reply:
Well its a permission check. We raise 404 if the org is not exportable ?

Reply:
The user will get a 403 if their org  is not exportable at all. This code comes to play only if you have org export_content_view permission and a search filter to go with it restricting it to only particular orgs. This behavior is identical to what we do in every controller.